spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseExpireSnapshotsSparkAction.java

```diff
@@ -147,7 +153,13 @@ public BaseExpireSnapshotsSparkAction deleteWith(Consumer<String> newDeleteFunc)
   public Dataset<Row> expire() {
     if (expiredFiles == null) {
       // fetch metadata before expiration
-      Dataset<Row> originalFiles = buildValidFileDF(ops.current());
+      Table staticTableBefore = newStaticTable(ops.current(), table.io());
+      allManifestsBefore = loadMetadataTable(staticTableBefore, ALL_MANIFESTS);
+      useCaching = PropertyUtil.propertyAsBoolean(options(), USE_CACHING, USE_CACHING_DEFAULT);
+      if (useCaching) {
+        allManifestsBefore.persist();
```

Comment:
I think we would want to do the "memory only" cache here (and in all the other usages) ... but I'm not sure